src/test/java/com/amplitude/api/DeviceInfoTest.java

```diff
@@ -304,4 +312,90 @@ public void testDontUseAdvertisingIdAsDeviceId() {
         assertNotEquals(advertisingId, client.getDeviceId());
         assertTrue(client.getDeviceId().endsWith("R"));
     }
+
+    @Test
+    public void testAppSetId() {
+        PowerMockito.mockStatic(AdvertisingIdClient.class);
+        String advertisingId = "advertisingId";
+        AdvertisingIdClient.Info info = new AdvertisingIdClient.Info(
+                advertisingId,
+                true
+        );
+        try {
+            Mockito.when(AdvertisingIdClient.getAdvertisingIdInfo(context)).thenReturn(info);
+        } catch (Exception e) {
+            fail(e.toString());
+        }
+
+        PowerMockito.mockStatic(UUID.class);
+        String mockUUIDString = "a74d2fb3-7c64-4d60-b0ca-b633aa7a4dbc";
+        UUID mockUUID = UUID.fromString(mockUUIDString);
+        try {
+            Mockito.when(UUID.randomUUID()).thenReturn(mockUUID);
+        } catch (Exception e) {
+            fail(e.toString());
+        }
+
+        String mockAppSetId = "5a8f0fd1-31a9-4a1f-bfad-cd5439ce533b";
+        PowerMockito.stub(PowerMockito.method(DeviceInfo.class, "getAppSetId"))
+                .toReturn(mockAppSetId);
+
+        Robolectric.getForegroundThreadScheduler().advanceTo(1);
+
+        AmplitudeClient client = Amplitude.getInstance("UseAppSetId");
+        client.useAppSetIdForDeviceId();
+        client.initialize(context, "1cc2c1978ebab0f6451112a8f5df4f4e");
+        ShadowLooper looper = Shadows.shadowOf(client.logThread.getLooper());
+        looper.runToEndOfTasks();
+
+        assertNotEquals(advertisingId, client.getDeviceId());
+        assertNotEquals(mockUUIDString, client.getDeviceId());
+        assertTrue(client.getDeviceId().endsWith("S"));
+    }
+
+    @Test
+    public void testDoNotSendAppSetId() {
+        String mockAppSetId = "5a8f0fd1-31a9-4a1f-bfad-cd5439ce533b";
+        PowerMockito.stub(PowerMockito.method(DeviceInfo.class, "getAppSetId"))
+                .toReturn(mockAppSetId);
+
+        ShadowLooper looper = Shadows.shadowOf(amplitude.logThread.getLooper());
+
+        amplitude.useAppSetIdForDeviceId();
+        amplitude.initialize(context, apiKey);
+        looper.runToEndOfTasks();
+        assertEquals(mockAppSetId + "S", amplitude.getDeviceId());
```

Comment:
Actually to this line in testDoNotSendAppSetId test case is basically what we need in last test case